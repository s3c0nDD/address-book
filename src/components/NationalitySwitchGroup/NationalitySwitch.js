import React from 'react';

import FormControlLabel from '@material-ui/core/FormControlLabel';
import Switch from '@material-ui/core/Switch';

const NationalitySwitch = ({ checked, handleChange, name }) => {
  const label = name.charAt(0).toUpperCase() + name.substring(1);

  return (
    <FormControlLabel
      control={
        <Switch
          checked={checked}
          onChange={handleChange}
          name={name}
        />
      }
      label={label}
    />
  );
};

export default NationalitySwitch;
