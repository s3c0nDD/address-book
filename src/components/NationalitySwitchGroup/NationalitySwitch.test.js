import React from 'react';
import { createMount, createShallow } from '@material-ui/core/test-utils';

import FormControlLabel from '@material-ui/core/FormControlLabel';
import Switch from '@material-ui/core/Switch';

import NationalitySwitch from './NationalitySwitch';

describe('NationalitySwitch', () => {
  let mount;
  let shallow;
  let component;
  let wrapper;
  let handleChangeMock;

  beforeAll(() => {
    shallow = createShallow();
    mount = createMount();
  });

  beforeEach(() => {
    handleChangeMock = jest.fn();
    component = (
      <NationalitySwitch
        checked={false}
        handleChange={handleChangeMock}
        name="polish"
      />
    );
    wrapper = shallow(component);
  });

  it('shallowly renders and matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has correct value of label prop', () => {
    const label = wrapper.find(FormControlLabel).props().label;
    expect(label).toEqual('Polish');
  });

  it('calls change handler on click', () => {
    const wrapper = mount(component);
    wrapper.find(Switch).props().onChange();
    expect(handleChangeMock).toBeCalledTimes(1);
  });

});
