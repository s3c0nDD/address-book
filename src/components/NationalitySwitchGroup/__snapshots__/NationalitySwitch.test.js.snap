// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NationalitySwitch shallowly renders and matches snapshot 1`] = `
<WithStyles(ForwardRef(FormControlLabel))
  control={
    <ForwardRef(WithStyles)
      checked={false}
      name="polish"
      onChange={[MockFunction]}
    />
  }
  label="Polish"
/>
`;
